react-components/src/util/I18NWrapper.js

```diff
@@ -0,0 +1,13 @@
+/**
+ @author sriram
+ */
+import React from "react";
+export default function withI18N(WrappedComponent, intlData) {
+    return class extends React.Component {
+        render() {
+            return (
+                <WrappedComponent {...intlData} {...this.props}/>
+            );
+        }
+    }
+}
```

Comment:
I did mine slightly different.  I wanted to make sure each component doesn't have to worry about the I18N implementation.  I also export a helper function for getting the strings, so if we ever change our implementation to something other than `react-intl` or if `react-intl` changes we hopefully only change these 2 functions, instead of every component that uses it.  I also pass the I18N props into `i18nData`.  I read that it's a good idea for HOCs to ensure their prop names won't collide with the underlying component's prop names.

```
import React, {Component} from 'react';
import {
    IntlProvider,
    FormattedMessage
} from 'react-intl';

export function hasI18N(WrappedComponent, i18nData) {
    return class extends Component {

        render() {
            return (
                <IntlProvider locale='en' defaultLocale='en' messages={i18nData}>
                    <WrappedComponent i18nData={i18nData} {...this.props} />
                </IntlProvider>
            )
        }
    }
}

export function getMessage(id) {
    return (
        <FormattedMessage id={id}/>
    )
}
```

Then my components look like this:
```
import {getMessage} from '../../util/hasI18N';
...

return (
            <RaisedButton label={getMessage('saveSearchBtn')}/>
)
```